rclc/src/rclc/executor.c

```diff
@@ -1075,11 +1684,6 @@ _rclc_execute(rclc_executor_handle_t * handle)
     }   // switch-case
   }
 
-  // corresponding callback of this handle has been called => reset data_available flag here
```

Comment:
Maybe this can be backported to foxy & galactic @Acuadros95 @JanStaschulat

Reply:
created an issue https://github.com/ros2/rclc/issues/209